plugins/plugin-git/che-plugin-git-ext-git/src/main/java/org/eclipse/che/ide/ext/git/client/compare/ComparePresenter.java

```diff
@@ -75,45 +77,106 @@ public ComparePresenter(AppContext appContext,
     }
 
     /**
-     * Show compare window.
+     * Show compare window for given set of files between given revision and HEAD.
      *
-     * @param file
-     *         file name with its full path
-     * @param status
-     *         status of the file
+     * @param alteredFiles
+     *         ordered touched files
+     * @param currentFile
+     *         file which will be shown first, if null then the first from the list will be shown
      * @param revision
      *         hash of revision or branch
      */
-    public void showCompareWithLatest(final File file, final Status status, final String revision) {
-        this.comparedFile = file;
+    public void showCompareWithLatest(final AlteredFiles alteredFiles,
+                                      @Nullable final String currentFile,
+                                      final String revision) {
+        this.alteredFiles = alteredFiles;
         this.revision = revision;
+
```

Comment:
Because the first block is set of field which holds data which is provided by user and the second one is the field which is related to internal logic of the presenter.